usersync/ejector.go

```diff
@@ -0,0 +1,134 @@
+package usersync
+
+import (
+	"errors"
+	"math"
+	"time"
+)
+
+type Ejector interface {
+	Choose(uids map[string]UIDEntry) (string, error)
+}
+
+type OldestEjector struct {
+	nonPriorityKeys []string
+	PriorityGroups  [][]string
+	EjectPriority   bool
+}
```

Comment:
Yeah this is true that `OldestEjector` will choose the oldest UID, but I have this state boolean because the ejector needs to figure out if it's supposed to eject the oldest priority UID or the oldest non priority UID. So that's why I passed in the Priority related values with this ejector.

If we wanted to remove these values, maybe I could alter the `PriorityEjector` to compile the specific set of `uids` of type `map[string]UIDEntry`, and then pass that to the `OldestEjector`?